include/beast/websocket/impl/write.ipp

```diff
@@ -244,101 +168,335 @@ operator()(error_code ec, bool again)
+            auto const n =
+                clamp(d.remain, d.ws.wr_.buf_size);
+            auto const b =
+                buffer(d.ws.wr_.buf.get(), n);
+            buffer_copy(b, d.cb);
+            detail::mask_inplace(b, d.key);
             d.remain -= n;
-            detail::mask_inplace(mb, d.key);
-            // send payload
+            // Send parial payload
             if(d.remain == 0)
-                d.state = 99;
-            BOOST_ASSERT(d.ws.wr_block_ == &d);
+                d.state = do_upcall;
             boost::asio::async_write(
-                d.ws.stream_, mb, std::move(*this));
+                d.ws.stream_, b, std::move(*this));
+            return;
+        }
+
+        //----------------------------------------------------------------------
+
+        case do_mask_frag:
+        {
+            auto const n = clamp(
+                d.remain, d.ws.wr_.buf_size);
+            d.remain -= n;
+            d.fh.len = n;
+            d.fh.key = d.ws.maskgen_();
+            d.fh.fin = d.fin ? d.remain == 0 : false;
+            detail::prepare_key(d.key, d.fh.key);
+            auto const b = buffer(
+                d.ws.wr_.buf.get(), n);
+            buffer_copy(b, d.cb);
+            detail::mask_inplace(b, d.key);
+            detail::write<static_streambuf>(
+                d.fh_buf, d.fh);
+            // Send frame
+            d.state = d.remain == 0 ?
+                do_upcall : do_mask_frag + 1;
+            BOOST_ASSERT(! d.ws.wr_block_);
+            d.ws.wr_block_ = &d;
+            boost::asio::async_write(d.ws.stream_,
+                buffer_cat(d.fh_buf.data(), b),
+                    std::move(*this));
+            return;
+        }
+
+        case do_mask_frag + 1:
+            d.cb.consume(
+                bytes_transferred - d.fh_buf.size());
+            d.fh_buf.reset();
+            d.fh.op = opcode::cont;
+            BOOST_ASSERT(d.ws.wr_block_ == &d);
+            d.ws.wr_block_ = nullptr;
+            if(d.ws.rd_op_.maybe_invoke())
+            {
+                d.state = do_maybe_suspend;
+                d.ws.get_io_service().post(
+                    std::move(*this));
+                return;
+            }
+            d.state = d.entry;
+            break;
+
+        //----------------------------------------------------------------------
+
+        case do_deflate:
+        {
+            auto b = buffer(d.ws.wr_.buf.get(),
+                d.ws.wr_.buf_size);
+            auto const more = detail::deflate(
+                d.ws.pmd_->zo, b, d.cb, d.fin, ec);
+            d.ws.failed_ = ec != 0;
+            if(d.ws.failed_)
+                goto upcall;
+            auto const n = buffer_size(b);
+            if(n == 0)
+            {
+                // The input was consumed, but there
+                // is no output due to compression
+                // latency.
+                BOOST_ASSERT(! d.fin);
+                BOOST_ASSERT(buffer_size(d.cb) == 0);
```

Comment:
Not yet. Right now, it's just a discrepancy. I haven't yet determined if it's a bug or not.